src/ngx_http_lua_socket_tcp.c

```diff
@@ -5230,6 +5230,10 @@ ngx_http_lua_socket_tcp_setkeepalive(lua_State *L)
          * so we increase connections here for backward compatibility.
          */
         spool->connections++;
+
+    } else if (n >= 3) {
+        ngx_log_error(NGX_LOG_WARN, pc->log, 0, "argument \"pool_size\" "
+                      "already specified in connect() method");
```

Comment:
@spacewander So you still want to support the pool_size argument in the `setkeepalive()` method?

Reply:
@agentzh 
Yes. The `pool_size` argument in `connect` method is experimental. Maybe it is better to be conservative.